seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/common/serialization/InternalRowConverter.java

```diff
@@ -80,6 +82,8 @@ private static Object convert(Object field, SeaTunnelDataType<?> dataType) {
                 return convertMap((Map<?, ?>) field, (MapType<?, ?>) dataType, InternalRowConverter::convert);
             case STRING:
                 return UTF8String.fromString((String) field);
+            case DECIMAL:
+                return Decimal.apply((BigDecimal) field);
```

Comment:
Please add `DECIMAL` type processing in `InternalRowConverter#reconvert(Object, SeaTunnelDataType<?>)`;

Reply:
Done.